### ISSUE TITLE
Some types of serialized data can prevent re-exporting from local

### ISSUE THREAD

**Issue by bobbingwide** [deliciousbrains/wp-migrate-db#70]
![image](https://cloud.githubusercontent.com/assets/2474435/23483707/f8ba5f44-fecb-11e6-93bd-3977d4d4531e.png)

### Scenario
1. Once upon a time google-sitemap-generator was used to generate an XML sitemap.
2. It created the above entry in the wp_options table
3. I believe I exported the live site to a local system using an SQL dump from cPanel
4. then imported manually.
5. When I tried to export the site back to live, using wp-migrate-db, I got a 500 internal server error.
This could be similar to https://wordpress.org/support/topic/migration-failed-3/
6. Running `wp migratedb export export-20170301` also failed
7. I found two rows in my `wp_options` table which, when print_r()ed, displayed
`Catchable fatal error: Object of class __PHP_Incomplete_Class could not be converted to string` 
8. I didn't need either so deleted them.

### Workarounds
1. Delete the offending rows from wp_options
2. Reactivate google-sitemap-generator to recreate the `sm_status` entry.

### Comments on the code
- WPMDB->is_json() calls json_decode(). If it's successful then json_decode is called again. 
- This seems inefficient. 
- Couldn't is_json() return the array from json_decode()? 
- is_json() should perform more tests before attempting to call json_decode().
- I'd be happier if the @ was removed from the call.

### Notes on my attempts to further analyse the problem
- I'm having difficulty recreating the invalid characters in my local MySQL database.
- The rows that caused problems were:
```
(2492,'sm_status','O:28:\"GoogleSitemapGeneratorStatus\":4:{s:39:\"\0GoogleSitemapGeneratorStatus\0startTime\";d:1430982865.7435801029205322265625;s:37:\"\0GoogleSitemapGeneratorStatus\0endTime\";d:1430982866.094378948211669921875;s:41:\"\0GoogleSitemapGeneratorStatus\0pingResults\";a:2:{s:6:\"google\";a:5:{s:9:\"startTime\";d:1430982865.784204006195068359375;s:7:\"endTime\";d:1430982865.904530048370361328125;s:7:\"success\";b:1;s:3:\"url\";s:110:\"http://www.google.com/webmasters/sitemaps/ping?sitemap=http%3A%2F%2Fwww.bobbingwidewebdesign.com%2Fsitemap.xml\";s:4:\"name\";s:6:\"Google\";}s:4:\"bing\";a:5:{s:9:\"startTime\";d:1430982865.922090053558349609375;s:7:\"endTime\";d:1430982866.0639240741729736328125;s:7:\"success\";b:1;s:3:\"url\";s:103:\"http://www.bing.com/webmaster/ping.aspx?siteMap=http%3A%2F%2Fwww.bobbingwidewebdesign.com%2Fsitemap.xml\";s:4:\"name\";s:4:\"Bing\";}}s:38:\"\0GoogleSitemapGeneratorStatus\0autoSave\";b:1;}','no')
```
where the \0's indicate the placement of the invalid character.

Also, this record produced the Warning message when the @ was removed from the json_decode() call.
```
(8245,'redirection_module_cache','a:1:{i:404;a:1:{i:0;O:15:\"Error404_Module\":5:{s:7:\"log_404\";b:1;s:2:\"id\";s:1:\"3\";s:4:\"type\";s:3:\"404\";s:4:\"name\";s:3:\"404\";s:7:\"options\";s:0:\"\";}}}
```

**Comment by bobbingwide** [comment]
Well there you go.... I fixed the problem in local, imported the new export into Live, got everything working, went to perform another export and got this.
![image](https://cloud.githubusercontent.com/assets/2474435/23486474/41939760-fed8-11e6-9563-bfbf2115ba29.png)

I honestly didn't expect that!

**Comment by bobbingwide** [comment]
The problem occurs when the oik-bwtrace plugin is tracing AJAX requests and is configured to report Error, Warning and Notice messages. I believe it's an unexpected message from the tracing code. I'll confirm this later.

**Comment by bobbingwide** [comment]
I can confirm that the reason for the Migration failed error message was due to the unexpected output from oik-bwtrace when it detected the Warning message being produced from json_decode().
While the underlying problems I mentioned in the comments still exist, I can also confirm that
1. The problem wasn't anything to do with invalid characters
2. I have fixed the problem in https://github.com/bobbingwide/oik-bwtrace/issues/56
